src/main/java/hudson/plugins/git/GitStatus.java

```diff
@@ -313,6 +325,54 @@ public void writeBody(PrintWriter w) {
         }
 
         /**
+         * Check if there is a ParameterValue with the same name
+         *
+         */
+        private boolean parameterValueExist (List<ParameterValue> parameterValues, String name) {
+            for (ParameterValue parameterValue : parameterValues) {
+                if (parameterValue.getName().equals(name)) {
+                    return true;
+                }
+            }
+            return false;
+        }
+
+        /**
+         * Get the default parameters values from a job
+         *
+         */
+        private List<ParameterValue> getDefaultParametersValues(Job job) {
+            ArrayList<ParameterValue> defValues;
+            ParametersDefinitionProperty paramDefProp = ((Job<?,?>)job).getProperty(ParametersDefinitionProperty.class);
+
+            if (paramDefProp != null) {
+                List <ParameterDefinition> parameterDefinition = paramDefProp.getParameterDefinitions();
+                defValues = new ArrayList<ParameterValue>(parameterDefinition.size());
+
+            } else {
+                defValues = new ArrayList<ParameterValue>(0);
```

Comment:
Seems wrong to use size 0 here, especially as there is a high probability of adding elements, perhaps have an int parameter that records the number of extra entries expected **or** say "screw this for a game of ___" and use the default constructor as I had originally hinted